File: QE-Java-Selenium/QE-selenium-java/javaselenium/src/main/java/javaselenium/demos/Activity12.java
package javaselenium.demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity12 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-content");
        System.out.println("Page Title: " + driver.getTitle());

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
        Actions actions = new Actions(driver);

        By clickMeBtn = By.xpath("//*[@id=\"genButton\"]");
        WebElement btn = wait.until(ExpectedConditions.visibilityOfElementLocated(clickMeBtn));
        actions.moveToElement(btn).click().perform();

        // Wait for ANY visible element that contains "Release"
        By releaseText = By.id("word");
        wait.until(ExpectedConditions.textToBePresentInElementLocated(releaseText,"release"));

        System.out.println("Release text found: " + driver.findElement(releaseText).getText());

        driver.quit();
    }
}

File: QE-Python_selenium/qe-selenium-python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/dynamic-content")
print("Page Title:", driver.title)

wait = WebDriverWait(driver, 15)
actions = ActionChains(driver) 
btn = wait.until(EC.visibility_of_element_located((By.ID, "genButton")))
actions.move_to_element(btn).click().perform()
 
word_el = wait.until(EC.visibility_of_element_located((By.ID, "word")))
wait.until(lambda d: "release" in word_el.text.lower())

print("Release text found:", word_el.text)

driver.quit()